Controls/README.md

```diff
@@ -1,40 +1,69 @@
+```
+sudo ./rtprio-setting.sh
+```
 
-1. Develop code in ```MotorController.c``` that will send the appropriate message across the CAN bus in a simulated fashion. This means it will write to the CAN.csv file with the following format: ```0xCAN_ID, CAN_MESSAGE```. It should accept both the ```CAN_ID``` and the ```CAN_MESSAGE``` as parameters. 
+Now, reboot your linux VM. Make sure your realtime priority limit has been set with:
+```
+ulimit -r
+```
+This should return *unlimited*. If it does not, ask someone for help or try the script again.
 
-2. Use the ```MotorController.h``` file to expose that written API by writing the function prototypes and then #including it in ```main.c``` and ```MotorController.c```.
+Create a branch for yourself after cloning the repository.
+```
+git checkout -b <your name>
+```
+All of your work will be on this branch.
 
-3. Create a main loop in ```main.c``` that calls the function written in ```MotorController.c``` at a set time interval (the main loop will be an infinite while loop). There must be some sort of delay (it can just be a simple for loop that does nothing for a lot of iterations). The specific amount of time that delay comprises will vary based on machine so that is not important.
+## The Mission
+In this starter project, you will be writing a mailbox and a cafe in the simulation version of the same Real-Time Operating System (RTOS) that runs on the car: the Micrium uCOS-III.
 
-4. Create a ```Makefile``` that compiles both the C files using ```gcc```. It should create object files and an executable (all of which you should add to the ```.gitignore```).
+### Mailbox
+The mailbox consists of the mailbox flag and the mailbox contents, both declared in ```Tasks.h```.
+1. Write the Mailman task in ```Mailman.c``` to put a unique message in the mailbox. The mailbox is simply a string (character array). Use the mailbox flag semaphore to prevent simultaneous access of the mailbox.
+2. Write the Recipient task in ```Recipient.c``` to print out the unique message in the mailbox. Use the mailbox flag semaphore to prevent simultaneous access of the mailbox.
+3. Write the test file ```Tests/Test_Mail.c```. This test file should create any semaphores, tasks, or variables necessary to run both the Mailman and Recipient tasks.
+4. To run the code, first compile with:
+	```
+	make all TEST=Mail
+	```
+	If this does not work or if you run into any issues, try ```make clean``` (this will remove any object files you just compiled) and recompile.
 
-5. Write a python script that reads from the CAN.csv file and displays the ID and message to the console in the following format: ```ID: 0x... Message: ...``` on an infinite loop. 
+### Cafe
+The cafe consists of the menu, item costs, and revenue, as well as the register occupied mutex. These are all declared in ```Tasks.h```
```

Comment:
```suggestion
The cafe consists of the menu, item costs, and revenue, as well as the register occupied mutex. These are all declared in ```Cafe.h```
```
Might be good to teach them naming that is actually good. I would *never* put a bunch of random shit in some file that is descriptively called "Tasks.h"

Reply:
Makes sense. I'll organize the structure a bit.